### ISSUE TITLE
Configuration option for '--ext' switch

### ISSUE THREAD

**Issue by bbirand** [eslint/eslint#3469]
I noticed that `eslint` wasn't finding my `.jsx` files. I found out about the `--ext` switch from the command-line, and I run it this way:
```bash
$ eslint --ext .js --ext .jsx
```

My questions are:
1. I couldn't find the configuration option for doing this from `.eslintrc`.
2. Shouldn't this be done by default when using the `jsx` features?
3. (why not just use `--ext js` instead of `--ext .js`, since these are by definition extensions?)

I'm using eslint v1.2.1.

Thanks!

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by lo1tuma** [comment]
1. You can’t define this in `.eslintrc` files because file traversal happens before the configuration is read
2. Same argument as in 1. at the time we do the file traversal we don’t if the `jsx` ecma feature is configured.

I think your command should be `eslint --ext .js --ext .jsx .` (you forgot the `'.'` which specifies the directory where eslint should start search for files).

**Comment by merlinpatt** [comment]
@bbirand You may have discovered this by now, but I had this issue too and found that you can do `js` and `jsx` without the dot. You can also do `eslint --ext=js,jsx .` to shorten the command a little more.

@lo1tuma I know this old and it's not going to be fixed but I am wondering why / how file traversal happens before reading the configuration. If the configuration is read later, how are rules applied?

**Comment by platinumazure** [comment]
@merlinpatt In order to resolve the correct configuration for a file, we need to read configuration in the same directory (and ancestor directories and maybe the user's home directory, and that doesn't even speak to configurations that extend other configuration files or packages). Thus, configurations are per-file and necessarily change as folders are traversed. However, extension specification affects traversal. So we need extensions specified outside of configuration files, because we don't read configuration files except while traversing.

The [CLIEngine Node API](http://eslint.org/docs/developer-guide/nodejs-api#cliengine) also allows extensions to be specified in its constructor, so you can use that if you are writing an application that needs to invoke ESLint programmatically and don't want to use the CLI itself.
